### PR TITLE
fix(devops): make pre-push path evidence completeness explicit

### PR DESCRIPTION
**Pull request by qnbs**
## Purpose

`resolvePushEvidence()` (introduced by #494) returns an identical empty
`changedFiles` array both for a genuine no-op push and for a tag-only push
that never produced real path proof — downstream local-admission consumers
could not tell a safe `NO_CHANGES` apart from incomplete evidence.

## Scope

Adds `pathEvidenceState: 'COMPLETE' | 'PARTIAL'` to `PushEvidence`:

- `PARTIAL` if any update in the push has `disposition === 'TAG'` (including
  mixed branch+tag pushes) — conservative, does not invent a changed-path
  set for tags
- `PARTIAL` on the `INVALID`/error path
- `COMPLETE` for branch/new-branch/deletion updates and for empty update sets

## Non-goals

- no change to local-admission classifier routing (`scripts/ci-prepush-*`)
- no consumption of `pathEvidenceState` by any caller yet (S1's job, not S3a's)

## Validation

- `pnpm exec vitest run tests/unit/signing.test.ts` — 19/19 passed (4 new
  cases: complete branch/new-branch/deletion, partial lightweight+annotated
  tag, partial mixed branch+tag, empty+invalid edge cases)
- `pnpm exec biome check` on touched files — clean
- `pnpm run typecheck` — clean

## Summary by Sourcery

Distinguish complete changed-path evidence from partial push evidence to enable safer downstream pre-push decisions.

New Features:
- Add explicit path-evidence completeness status to push evidence results, distinguishing complete changed-file proof from partial evidence.

Bug Fixes:
- Prevent tag-only and mixed branch/tag pushes from being misclassified as confirmed no-change results.

Tests:
- Add coverage for complete branch updates, partial tag and mixed updates, empty evidence, and invalid evidence cases.

### CODE REVIEW

File: scripts/signing/signing-core.mjs [hunk 1]
```diff
@@ -366,12 +366,22 @@ export function resolvePushEvidence(input, cwd = process.cwd(), dependencies = {
         disposition: isZeroSha(update.remoteSha) ? 'NEW_BRANCH' : 'UPDATED',
       });
     }
-    return { updates: evidenceUpdates, changedFiles: [...changedFiles], evidenceState: 'RESOLVED' };
+    // QNBS-v3: tag updates prove object validity but not a complete changed-path set.
+    const pathEvidenceState = evidenceUpdates.some(({ disposition }) => disposition === 'TAG')
```
[the remark this answers is not in the document]
Comment (qnbs) [reply scripts/signing/signing-core.mjs:370]: Investigated — I believe this is a false positive, not a fix.

A tag *deletion* (`localSha` all-zero) hits `isZeroSha(update.localSha)` first and is classified
`DELETED` before the `refs/tags/` check ever runs, so it's correctly excluded from the
`.some(disposition === 'TAG')` check and reported `COMPLETE` — same as a branch deletion. This is
intentional, not an oversight: a deletion introduces zero new commit content, so there is nothing
for `pathEvidenceState` to be uncertain about. The `PARTIAL` classification exists specifically for
tag *creates/updates*, where a tag can point at a commit whose content was never diffed against a
known base (the actual release-significance risk this field guards against, since `v*` tags can
trigger `ci.yml`/`docker.yml`/`tauri-build.yml`'s release/build jobs).

Verified this repo's tag-triggered workflows all use `on: push: tags: [...]` (`ci.yml:13`,
`docker.yml:8`, `tauri-build.yml:9`) — per GitHub Actions semantics, a ref *deletion* fires a
`delete` event, not `push`, so none of them can run from a tag deletion. There is no unproven
content and no release-significance exposure for the deletion case this comment flags.

The existing test `reports complete path evidence for branch, new-branch, and deletion updates`
already covers deletion (generically, not tag-specific) returning `COMPLETE`, confirming this is
the deliberate, tested design, not a gap. Leaving as-is; resolving this thread.

**Comment by qnbs** [comment]
@coderabbitai review

**Comment by qnbs** [comment]
@codeant-ai: review
